File: BS/soluciones/go/solve.go
package main

import (
    "fmt"
    "log"
    "math/big"
    "os"
    "time"
)

func main() {
    // Medir tiempo de inicio
    inicio := time.Now()

    // Definimos las variables como punteros a big.Int
    f0 := big.NewInt(0) // Fibonacci(0)
    f1 := big.NewInt(1) // Fibonacci(1)
    suma := big.NewInt(0)

    // Iteramos 10,000 veces
    for i := 0; i < 10000; i++ {
        // Sumamos el término actual (f0)
        suma.Add(suma, f0)

        // Calculamos el siguiente número de Fibonacci
        var siguiente big.Int
        siguiente.Add(f0, f1)

        // Avanzamos en la sucesión
        f0.Set(f1)
        f1.Set(&siguiente)
    }

    // Medir tiempo de finalización
    duracion := time.Since(inicio).Milliseconds()

    // Imprimimos la suma total en consola
    fmt.Printf("La suma de los primeros 10,000 números de Fibonacci es: %s\n", suma.String())
    fmt.Printf("Tiempo transcurrido: %d ms\n", duracion)

    // Crear (o sobrescribir) el archivo de salida
    file, err := os.Create("output.txt")
    if err != nil {
        log.Fatalf("Error creando el archivo: %v", err)
    }
    defer file.Close()

    // Escribimos en el archivo la suma y el tiempo de ejecución
    _, err = fmt.Fprintf(file, "La suma de los primeros 10,000 números de Fibonacci es: %s\n", suma.String())
    if err != nil {
        log.Fatalf("Error escribiendo en el archivo: %v", err)
    }

    _, err = fmt.Fprintf(file, "Tiempo transcurrido: %d ms\n", duracion)
    if err != nil {
        log.Fatalf("Error escribiendo en el archivo: %v", err)
    }
}

File: BS/soluciones/java/solve.java
import java.math.BigInteger;
import java.io.FileWriter;
import java.io.IOException;

public class solve {

    public static BigInteger sumaFibonacci(int n) {
        BigInteger a = BigInteger.ZERO; // F(0)
        BigInteger b = BigInteger.ONE; // F(1)
        BigInteger total = BigInteger.ZERO;

        for (int i = 0; i < n; i++) {
            total = total.add(a);
            BigInteger temp = a;
            a = b;
            b = b.add(temp);
        }

        return total;
    }

    public static void main(String[] args) {
        int n = 10000;
        long startTime = System.currentTimeMillis();
        BigInteger resultado = sumaFibonacci(n);
        long endTime = System.currentTimeMillis();
        long tiempoEjecucion = endTime - startTime;

        // Imprimimos en consola
        System.out.println("La suma de los primeros " + n + " términos de Fibonacci es: " + resultado);
        System.out.println("Tiempo de ejecución: " + tiempoEjecucion + " ms");

        // Escribimos los resultados en un archivo de texto
        try (FileWriter writer = new FileWriter("output.txt")) {
            writer.write("La suma de los primeros " + n + " términos de Fibonacci es: " + resultado + "\n");
            writer.write("Tiempo de ejecución: " + tiempoEjecucion + " ms\n");
        } catch (IOException e) {
            System.err.println("Error al escribir en el archivo: " + e.getMessage());
        }
    }
}
